src/main/java/socket/BidWS.java

```diff
@@ -353,6 +379,155 @@ protected void onAuctionBid(BodyWS body) {
         auctionBidded(dbBid);
     }
 
+    private void onCombinatorialAuctionBid(BodyWS body, Bid[] unsafeBids, Auction dbAuction, int userId) {
+        BidDAO bidDAO = BidDAOSQL.getInstance();
+        UserDAO userDAO = UserDAOSQL.getInstance();
+        GoodDAO goodDAO = GoodDAOSQL.getInstance();
+
+        List<Bid> unsafeBidsList = new ArrayList<>(Arrays.asList(unsafeBids));
+
+        // 1. All bids must have the same auction ID and amount.
+        double commonAmount = Math.floor(unsafeBids[0].amount * 100) / 100;
+        int commonAuctionId = unsafeBids[0].auctionId;
+        Set<Integer> differentGoods = new HashSet<>();
+        for (Bid unsafeBid: unsafeBidsList) {
```

Comment:
For most checks I think you didn't really need the loop.
As long as the first one carried the right information, you could ignore the rest.

Reply:
@etorres404 yeah, it's fine